src/components/api.js

```diff
@@ -0,0 +1,94 @@
+async function getActors() {
+  return await apiCall('GET', 'actors');
+}
+
+async function deleteActor(id) {
+  return await apiCall('DELETE', `actors/${id}`);
+};
+
+async function newActor(name) {
+  name = JSON.stringify({ name });
+  return await apiCall('POST', 'actors', name, { 'Content-Type': 'application/json' });
+}
+
+async function getCurrentActorCharacters(actor, currentScript) {
+  return await apiCall('GET', `script/${currentScript}/characters?actorId=${actor.id}`);
+}
+
+async function getCharacters(currentScript) {
+  return await apiCall('GET', `script/${currentScript}/characters`);
+}
+
+async function getScript(scriptId) {
+  return await apiCall('GET', 'script/' + scriptId);
+}
+
+async function getScriptTitles() {
+  return await apiCall('GET', 'scripts/title');
+}
+
+async function deleteScript(deleteId) {
+  return await apiCall('DELETE', `script/${deleteId}`);
+}
+
+async function uploadScript(formData) {
+  return await apiCall('POST', 'script', formData);
+}
+
+async function postPosition(actorId, scriptId, position) {
+  const body = JSON.stringify({ actorId, scriptId, position });
+  return await apiCall('POST', 'positions', body, { 'Content-Type': 'application/json' });
+}
+
+async function apiCall(method, uri, body = null, headers = {}) {
+
+  try {
+    const response = await fetch('/api/' + uri, {
+      method: method,
+      headers: headers,
+      body: body
+    });
+    // Set data for successful fetch
+    if (response.ok) {
+      if (method === 'DELETE') {
+        return {
+          success: true,
+          data: 'Deleted'
```

Comment:
Instead of adding this extra data here, actually return the deleted item in the backend.